storage/remote/write_handler_test.go

```diff
@@ -78,6 +86,32 @@ func TestOutOfOrder(t *testing.T) {
 	require.Equal(t, http.StatusBadRequest, resp.StatusCode)
 }
 
+// This test case currently aims to verify that the WriteHandler endpoint
+// don't fail on ingestion errors since the exemplar storage is
+// still experimental.
+func TestOutOfOrderExemplar(t *testing.T) {
+	buf, _, err := buildWriteRequest([]prompb.TimeSeries{{
+		Labels:    []prompb.Label{{Name: "__name__", Value: "test_metric"}},
+		Exemplars: []prompb.Exemplar{{Labels: []prompb.Label{{Name: "foo", Value: "bar"}}, Value: 1, Timestamp: 0}},
+	}}, nil, nil)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest("", "", bytes.NewReader(buf))
+	require.NoError(t, err)
+
+	appendable := &mockAppendable{
+		latestExemplar: 100,
+	}
+	handler := NewWriteHandler(log.NewNopLogger(), appendable)
+
+	recorder := httptest.NewRecorder()
+	handler.ServeHTTP(recorder, req)
+
+	resp := recorder.Result()
+	// TODO: update to require.Equal(t, http.StatusNoContent, resp.StatusCode) once the exemplar storage is no more experimental.
```

Comment:
Did you mean `http.StatusConflict` or some other error here? It looks the same as the current code right now.

Nit: `once exemplar storage is not experimental` instead of no more.

Reply:
updated with suggestion